pkg/istio-agent/agent.go

```diff
@@ -225,9 +237,92 @@ func (a *Agent) initializeEnvoyAgent() error {
 
 	drainDuration, _ := types.DurationFromProto(a.proxyConfig.TerminationDrainDuration)
 	a.envoyAgent = envoy.NewAgent(envoyProxy, drainDuration)
+	a.envoyWaitCh = make(chan error, 1)
+	if a.cfg.EnableDynamicBootstrap {
+		// Simulate an xDS request for a bootstrap
+		go func() {
+			// wait indefinitely and keep retrying with jittered exponential backoff
+			backoff := 500
```

Comment:
the backoff logic seems wrong:
```
2021-05-07T14:42:27.842702Z     info    retrying bootstrap discovery request with backoff: 183ms
2021-05-07T14:42:28.026717Z     info    retrying bootstrap discovery request with backoff: 32ms
2021-05-07T14:42:28.059484Z     info    retrying bootstrap discovery request with backoff: 63ms
2021-05-07T14:42:28.123340Z     info    retrying bootstrap discovery request with backoff: 91ms
2021-05-07T14:42:28.215286Z     info    retrying bootstrap discovery request with backoff: 48ms
2021-05-07T14:42:28.264136Z     info    retrying bootstrap discovery request with backoff: 25ms
2021-05-07T14:42:28.289938Z     info    retrying bootstrap discovery request with backoff: 47ms
2021-05-07T14:42:28.337921Z     info    retrying bootstrap discovery request with backoff: 73ms
2021-05-07T14:42:28.411752Z     info    retrying bootstrap discovery request with backoff: 8ms
2021-05-07T14:42:28.420661Z     info    retrying bootstrap discovery request with backoff: 5ms
2021-05-07T14:42:28.426413Z     info    retrying bootstrap discovery request with backoff: 7ms
2021-05-07T14:42:28.434449Z     info    retrying bootstrap discovery request with backoff: 9ms
2021-05-07T14:42:28.444404Z     info    retrying bootstrap discovery request with backoff: 1ms
2021-05-07T14:42:28.445972Z     info    retrying bootstrap discovery request with backoff: 0s
panic: runtime error: integer divide by zero

goroutine 98 [running]:
istio.io/istio/pkg/istio-agent.(*Agent).initializeEnvoyAgent.func1(0xc0014d8a80, 0xc000248540, 0x45bd0b8, 0xc0015820b0)
        /home/howardjohn/go/src/istio.io/istio/pkg/istio-agent/agent.go:263 +0x23e
created by istio.io/istio/pkg/istio-agent.(*Agent).initializeEnvoyAgent
        /home/howardjohn/go/src/istio.io/istio/pkg/istio-agent/agent.go:243 +0x72b
```

Reply:
Why did any test not fail with this back off ?

Reply:
Sorry, made an error transcribing from C++ (fighting int to duration silliness). This should be correct now.
The logic is lifted straight from Envoy for compatibility with other xDS clients.

Reply:
Tests did not fail because it happens rarely in integrations.